tests/Unit/DataStructures/Test_ScaleBy.cpp

```diff
@@ -0,0 +1,87 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include <catch.hpp>
+
+#include "DataStructures/Tensor/EagerMath/ScaleBy.hpp"
+
+TEST_CASE("Unit.DataStructures.Tensor.ScaleBy", "[Functors][Unit]") {
+  const size_t npts = 8;
+  DataVector one(npts, 1.0);
+  DataVector two(npts, 2.0);
+  DataVector three(npts, 3.0);
+  DataVector four(npts, 4.0);
+  DataVector five(npts, 5.0);
+  DataVector twelve(npts, 12.0);
+  DataVector thirteen(npts, 13.0);
+
+  Tensor<DataVector, Symmetry<1>,
+         typelist<SpatialIndex<1, UpLo::Lo, Frame::Grid>>>
+      one_d_covector;
+  one_d_covector.get<0>() = two;
+  DataVector magnitude_one_d_covector = two;
+  auto normalized_one_d_covector =
+      scale_by(one_d_covector, magnitude_one_d_covector);
+  for (size_t s = 0; s < npts; ++s) {
+    CHECK(normalized_one_d_covector.get<0>()[s] == 1.0);
+  }
+
+  Tensor<DataVector, Symmetry<1>,
+         typelist<SpacetimeIndex<1, UpLo::Up, Frame::Grid>>>
+      two_d_vector;
+  two_d_vector.get<0>() = three;
+  two_d_vector.get<1>() = four;
+  DataVector magnitude_two_d_vector = five;
+  auto normalized_two_d_vector = scale_by(two_d_vector, magnitude_two_d_vector);
+  for (size_t s = 0; s < npts; ++s) {
+    CHECK(normalized_two_d_vector.get<0>()[s] == 0.6);
+    CHECK(normalized_two_d_vector.get<1>()[s] == 0.8);
+  }
+
+  Tensor<DataVector, Symmetry<1>,
+         typelist<SpatialIndex<2, UpLo::Up, Frame::Grid>>>
+      two_d_spatial_vector;
+  two_d_spatial_vector.get<0>() = five;
+  two_d_spatial_vector.get<1>() = twelve;
+  DataVector magnitude_two_d_spatial_vector = thirteen;
+  auto normalized_two_d_spatial_vector =
+      scale_by(two_d_spatial_vector, magnitude_two_d_spatial_vector);
+  for (size_t s = 0; s < npts; ++s) {
+    CHECK(normalized_two_d_spatial_vector.get<0>()[s] == 5.0 / 13.0);
+    CHECK(normalized_two_d_spatial_vector.get<1>()[s] == 12.0 / 13.0);
+  }
+
+  Tensor<DataVector, Symmetry<1>,
+         typelist<SpatialIndex<3, UpLo::Lo, Frame::Grid>>>
+      three_d_covector;
+  three_d_covector.get<0>() = three;
+  three_d_covector.get<1>() = twelve;
+  three_d_covector.get<2>() = four;
+  DataVector magnitude_three_d_covector = thirteen;
+  auto normalized_three_d_covector =
+      scale_by(three_d_covector, magnitude_three_d_covector);
+  for (size_t s = 0; s < npts; ++s) {
+    CHECK(normalized_three_d_covector.get<0>()[s] == 3.0 / 13.0);
+    CHECK(normalized_three_d_covector.get<1>()[s] == 12.0 / 13.0);
+    CHECK(normalized_three_d_covector.get<2>()[s] == 4.0 / 13.0);
+  }
+
+  Tensor<DataVector, Symmetry<1>,
+         typelist<SpacetimeIndex<4, UpLo::Lo, Frame::Grid>>>
+      five_d_covector;
```

Comment:
There is a constructor for `tnsr::vector` that takes a list of `DataVector`s that could be used here. You may need three braces, unfortunately, to get it to work.